src/CoreBundle/Service/CmsConfigDataAccess.php

```diff
@@ -0,0 +1,58 @@
+<?php
+
+/*
+ * This file is part of the Chameleon System (https://www.chameleonsystem.com).
+ *
+ * (c) ESONO AG (https://www.esono.de)
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace ChameleonSystem\CoreBundle\Service;
+
+use Doctrine\DBAL\Connection;
+use Doctrine\DBAL\Exception;
+
+class CmsConfigDataAccess implements CmsConfigDataAccessInterface
+{
+    /**
+     * @var Connection
+     */
+    private $connection;
+
+    public function __construct(Connection $connection)
+    {
+        $this->connection = $connection;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public function getBackendTheme(): ?\TdbPkgCmsTheme
+    {
+        // TODO ? use a runtime-cache; this might be called several times...
```

Comment:
please check if it is called multiple times. If so, add a runtime cache decorator.

Reply:
Well it is. Normally 3 times.

However I don't get a cache decorator working (is always non-shared).

Reply:
It seems to be actually independent requests. So a runtime cache is not necessary /wouldn't work.